### DISCUSSION TITLE
Sampling images at any (smaller) size

### DISCUSSION THREAD

**Discussion by YotamNitzan** [PixArt-alpha/PixArt-alpha#30] [Q&A] [answered]
Is it possible to generate small images from the 1024x1024 multi-scale model? For example 256x256.
I naively tried feeding such values to the size embedder or simply use a smaller latent, which work for sizes similar to 1024x1024 but breaks for much smaller images. 

Do you have an idea of how to do that?

**Comment by lawrence-cj** [comment, marked as answer]
Nope, the 1024px model can only generate 1K resolution images within multiple scales, due to the training process and the position embeddings. We provide 512px model for further development. If you want to generate smaller size images, e.g. 256px, you can train from 512 with lewei_scale=256/512=0.5 ([this variable is a trick for upsample or downsample training](https://github.com/PixArt-alpha/PixArt-alpha/blob/cee9c6be8050c99d91422528d3ebc970fdc70182/configs/PixArt_xl2_internal.py#L13)). In the future, we may also release the 256px model for the community.

**Reply by ruizhaocv** [reply]
Hi, thanks for the awesome work and easy-to-follow code. 
If I want to train a 512px model from a 256px model, do I need to set the lewei_scale=512/256=2? I found that, in the inference code, the lewei_scale of 256px and 512px models are both 1.0.

**Reply by lawrence-cj** [reply]
It depends on the lewei_scale you used in 256, it should be lewei_scale =512/256 * lewei_scale_used_in_256

**Reply by ruizhaocv** [reply]
Thanks for your quick reply. I used "PixArt-α-SAM-256" as the 256px model. How much is the lewei_scale of it? 
I think the lewei_scale in the three training stages should be 0.5 for 256px, 1.0 for 512px, and 2.0 for 1024px. Is that correct？

**Reply by ruizhaocv** [reply]
When inference with "PixArt-α-SAM-256" and "PixArt-α-256", I found that the 1.0 lewei_scale works well but the 0.5 lewei_scale leads to bad generation. So I believe when you pre-train the 256px models, the lewei_scale is set to 1.0. Why did the pre-trained 512px model also set the lewei_scale to 1.0 rather than 2.0?

**Reply by lawrence-cj** [reply]
Hh, since we haven't used the position interpolation at the very beginning. Then you should use lewei_scale=2.0 for 512 training.

**Reply by ruizhaocv** [reply]
Ah, I see. That clears up my confusion. How did you reuse the pre-trained 256px model to train the 512px model? Without position interpolation? What is the drawback of upscale training without position interpolation? More training steps or an unstable training process?

**Reply by lawrence-cj** [reply]
It just need a little bit more iterations to converge.

**Reply by ruizhaocv** [reply]
Thank you very much for your response.
